clustering/ejb/WFLY-14359_Create_distributable-ejb_subsystem.adoc

```diff
@@ -0,0 +1,170 @@
+= WFLY-14953 Create distributable-ejb subsystem
+:author:            Richard Achmatowicz
+:email:             rachmato@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The Undertow and EJB subsystems depend on certain abstractions provided by the clustering subsystems in order to support
+clustered deployments.
+
+Hard-coding clustering subsystem dependencies within the EJB and Undertow subsystems creates a tight-coupling between
+the subsystems in question, and also makes configuring the various dependencies (or plugging in alternative dependencies)
+difficult. Defining a separate "configuration" subsystem to sit in between the EJB/Undertow subsystem and the relevant clustering
+subsystems simplifies configuration and provides a looser coupling between the subsystems.
+
+The distributable-web subsystem was introduced to provide decoupling between the Undertow subsystem and the relevant clustering
+subsystems. The aim of the distriubutable-ejb subsystem is to do the same for the EJB subsystem and the clustering abstractions
+that it depends on.
+
+There are at present four features provided by the EJB3 subsystem which depend on clustering abstractions:
+
+* provision of cache factories, used to store stateful session bean sessions
+* provision of client mappings registries, used to provide EJB client topology updates
+* provision of clustered singletons, used to suppport message driven beans marked as singleton beans
+* provision of a persistent timer store, used to support EJB timers
+
+The distributable-ejb subsystem will allow defining different backend providers of these abstractions, which may be configured
+independently of the ejb3 subsystem.
+
+Ultimately, the distributable-ejb subsystem will provide providers for all four of the feature areas listed above.
+But this issue will be restricted to the following scope:
+
+* defining the distributable-ejb subsystem
+* defining a cache factory provider for stateful session beans and integrating that provider into the ejb3 subsystem
+* defining a client mappings registry provider for the EJB client and integrating that provider into the ejb3 subsystem
+
+The remaining two features will be completed in a separate RFE.
```

Comment:
WFLY-7628 will address this for distributed timer management.
We need to create a feature request jira for the singleton mdb delivery groups stuff.